include/seastar/core/app-template.hh

```diff
@@ -129,13 +129,15 @@ private:
     std::shared_ptr<smp> _smp;
     seastar_options _opts;
     boost::program_options::options_description _app_opts;
+    boost::program_options::options_description* _app_visible_opts_for_help = nullptr;
```

Comment:
i'd suggest use `&_app_opts` as the default value of `_app_visible_opts_for_help`, like:
```c++
boost::program_options::options_description* _app_visible_opts_for_help = &_app_opts;
```

this would help with the readability -- it's clear that we use `_app_opts` by default, when handling `--help`.